### DISCUSSION TITLE
reenable sntrup761

### DISCUSSION THREAD

**Discussion by seschu** [open-quantum-safe/oqs-provider#330] [Q&A] [answered]
I have noticed that sntrup761 is still part of libqs and that an old generate.yml file also inlcuded sntrup761. Is it still possible to manually add this algo to the oqs provider lib?

**Comment by seschu** [comment]
What I did sofar is I added this 

diff --git a/oqs-template/generate.yml b/oqs-template/generate.yml
index 099d36e..99940f1 100644
--- a/oqs-template/generate.yml
+++ b/oqs-template/generate.yml
\@\@ -1,6 +1,20 \@\@
 \# This is the master document for ID interoperability for KEM IDs, p-hybrid KEM IDs, SIG (O)IDs
 \# Next free plain KEM ID: 0x0247, p-hybrid: 0x2F47, X-hybrid: 0x2FB2
 kems:
\+  -
\+    family: 'NTRU-Prime'
\+    name_group: 'sntrup761'
\+    nid: '0x0233'
\+    nid_hybrid: '0x2F44'
\+    extra_nids:
\+      current:
\+        - hybrid_group: "x25519"
\+          nid: '0x2FB3'
\+      old:
\+        - implementation_version: Round 3 p384 hybrid
\+          nist-round: 3
\+          nid: '0x2F33'
\+    oqs_alg: 'OQS_KEM_alg_ntruprime_sntrup761'
  \-
     family: 'FrodoKEM'
     name_group: 'frodo640aes'


And then I executed python3 generate.py after setting LIBOQS_SRC_DIR.
Then everything seems to work sofar.

Is the assumption right this can then also be used in a TLS like session?

**Reply by baentsch** [reply, marked as answer]
Yes (to each of the separate questions above).

**Comment by dstebila** [comment]
Note that it would not be our intention to permanently re-enable the NTRU Prime suite.  We only keep strnup761 in liboqs because this specific algorithm is used in OpenSSH.  But since there are (to my knowledge) no standards bodies pursuing standardization of NTRU Prime, we would eventaully deprecate strnup761 in liboqs and remove it.

**Reply by baentsch** [reply]
ACK. Accordingly we probably wouldn't accept a PR into `oqsprovider` doing the above unless there's a compelling reason in light of the comments by @dstebila (read: Why in the world do you want to do this, @seschu ? :) But the approach to achieving what you want is correct.

**Reply by seschu** [reply]
It is just for having compatibility to some older projects I hav an interest in. Of course it is not necessary in the scope of this project to keep theese algorithms around, but right now it is pretty convenient that I can just reactivate the algorithm. Dont expect any PR from me because of this. 

BTW: Do you know the reason why libqs still supports this but oqs provider has remived it?

**Reply by seschu** [reply]
Sorry if this type of question is not the scope of the issue section...

**Reply by baentsch** [reply]
> BTW: Do you know the reason why libqs still supports this but oqs provider has remived it?

As stated by @dstebila there's only been a "backwards compatibilty" reason for this algorithm in oqs-openssh. Nobody showed interest in retaining it for TLS/openssl. Hence it was removed from oqsprovider.

**Reply by loganaden** [reply]
If I remember correctly, this is due to OpenSSH stating publicly that sntruprime would be supported as a long term KEX. @baentsch @dstebila With the current patent concerns that were mentioned in the IETF, it might be safer to use have sntrup761 enabled  for implementors that might have some concerns ?

**Reply by loganaden** [reply]
@dstebila There is this draft: https://datatracker.ietf.org/doc/draft-josefsson-ntruprime-ssh/

**Reply by baentsch** [reply]
I'm a bit confused as to why we have this `openssh`-related discussion in oqs-provider (only serving `openssl`): Can you please refresh my memory on this, @loganaden ?

**Reply by loganaden** [reply]
@baentsch agreed. I went off-topic there. Btw, would a PR to enable ntruprime for openssl be considered ? we want to offer both ML-KEM and sntruprime on a server.

**Reply by baentsch** [reply]
> Btw, would a PR to enable ntruprime for openssl be considered ?

I will consider anything coming with a good reason and an assertion to support it (see for example https://github.com/open-quantum-safe/oqs-provider/issues/351) (and passing CI of course :-)
